marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -228,9 +225,56 @@ handleQuery ::
   Query a ->
   HalogenM State Action ChildSlots Void m (Maybe a)
 handleQuery settings (ChangeRoute route next) = do
-  handleRoute settings route
+  -- Without the following each route is handled twice, once when we call selectView ourselves
+  -- and another which is triggered in Main, when the route changes.
+  currentView <- use _view
+  when (routeToView route /= Just currentView) $ handleRoute settings route
   pure $ Just next
 
+------------------------------------------------------------
+fullHandleAction ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action ->
+  HalogenM State Action ChildSlots Void m Unit
+fullHandleAction settings =
+  withAccidentalNavigationGuard settings
+    $ handleActionWithAnalyticsTracking
+        ( handleAction settings
+        )
+
+actionWithAnalytics ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action ->
+  HalogenM State Action ChildSlots Void m Unit
+actionWithAnalytics settings =
+  handleActionWithAnalyticsTracking
+    ( handleAction settings
+    )
+
+-- This is a helper HOF to log all the actions handled by the MainFrame. It is always commented
+-- so that it doesn't trigger a compilation warning (because of traceM). To use it, put it as the
+-- first action handler in fullHandleAction and actionWithAnalytics
+-- withLog ::
+--   forall m.
+--   MonadAff m =>
+--   String ->
+--   (Action -> HalogenM State Action ChildSlots Void m Unit) ->
+--   Action ->
+--   HalogenM State Action ChildSlots Void m Unit
+-- withLog handlerName handleAction' action = do
+--   traceM $ "MainFrame.State: " <> handlerName
+--   traceM action
+--   handleAction' action
+--
```

Comment:
I'd like to have it in but commented. It is useful to have it defined for debugging purposes but even if I don't use it, it generates a warning

Reply:
Just export it, then you won't get the warning. If you leave it commented out, it will rot, I guarantee.

Reply:
Tried exporting it but it doesn't remove the warning... I can keep it only in my machine as a snippet, but I think it could be useful to others to know how to debug the actions

Reply:
My issue is that the comment is linked to the documentation of `handleAction`, so I bet some documentation tool will show the whole thing when showing documentation for `handleAction`. I think it would be good to remove the dashes from lines 273 and 274, so that they are separate comment blocks :)

Reply:
If I remove those lines, purty brings the comments together. For the moment I'm removing it from the code and I have locally as a snippet